### PR TITLE
Migrate to rtnetlink

### PR DESCRIPTION
**Pull request by mxpv**
Since we got async/awaits with Tokio in (#1210), we now can switch to `rtnetlink` crate to manage netlink related tasks (see original discussion here #1171).

This would allow to get rid of low level code and rely on high level async API.
As a side effect, rtnetlink has full IPv6 support, so it'll address #1171

### CODE REVIEW

**Comment by mxpv** [comment]
Network configuration I got on my env:

Go Agent
```
/ # ip l
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9000 qdisc pfifo_fast qlen 1000
    link/ether 52:97:bc:02:f6:be brd ff:ff:ff:ff:ff:ff
/ # ip r
default via 10.14.60.98 dev eth0
10.14.60.96/28 dev eth0 scope link  src 10.14.60.104
/ # ip -6 r
2620:149:1007:1b21::/64 dev eth0  metric 256
fe80::/64 dev eth0  metric 256
ff00::/8 dev eth0  metric 256
```

Rust agent
```
/ # ip l
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9000 qdisc pfifo_fast qlen 1000
    link/ether 52:dd:70:86:52:86 brd ff:ff:ff:ff:ff:ff
/ # ip r
default via 10.14.60.98 dev eth0
10.14.60.96/28 dev eth0 scope link  src 10.14.60.106
/ # ip -6 r
2620:149:1007:1b21::/64 dev eth0  metric 256
fe80::/64 dev eth0  metric 256
default via 2620:149:1007:1b21::2 dev eth0  metric 1024
ff00::/8 dev eth0  metric 256
```

**Comment by mxpv** [comment]
CI looks promising.
(I'll update the header once I obtain an approval).

File: src/agent/src/netlink.rs
Comment (Tim-Zhang) [inline src/agent/src/netlink.rs:1]: Copyright missing.

**Comment by Tim-Zhang** [comment]
@mxpv Thanks a lot @yyyeerbo can you give a review?

File: src/agent/src/netlink2.rs
Comment (yyyeerbo) [inline src/agent/src/netlink2.rs]: We might need a more robust way to determine if it is ipv6. "::" indicates ipv6 address contains zeros? it might not be the case
Maybe we can try to parse ipaddr as v4and v6, to determine it is v4 or v6

Comment (mxpv) [reply src/agent/src/netlink2.rs]: Updated

**Comment by yyyeerbo** [comment]
LGTM

File: src/agent/src/netlink.rs
Comment (mxpv) [reply src/agent/src/netlink.rs]: Yep, I've updated the file with generic header (with no company name), just to pass checks and advance to unit tests. I'm still waiting for a legal approval.

**Review by yyyeerbo — APPROVED** [review]

**Comment by little-dude** [comment]
Glad to see rtnetlink being used in the wild :)

**Review by lifupan — APPROVED** [review]

**Comment by dgibson** [comment]
This merge appears to have broken the Rust agent unit tests.  I'm not sure how that wasn't caught before merge.

```
failures:

---- netlink::tests::list_interfaces stdout ----
thread 'netlink::tests::list_interfaces' panicked at 'assertion failed: `(left != right)`
  left: `0`,
 right: `0`', src/netlink.rs:876:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    netlink::tests::list_interfaces

test result: FAILED. 48 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```

**Pull request merged**
